### PR TITLE
TST: use `with` where possible instead of manual `close`

### PR DESCRIPTION
**Pull request by akx**
- [ ] closes #xxxx (Replace xxxx with the Github issue number)
  - Nothing to close – just noticed these while working on #48922.
- [x] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
  - Only tests changed. 
- [x] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [x] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
  - Nothing changed.
- [x] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.
  - Not done (not required)

### CODE REVIEW

**Comment by mroeschke** [comment]
Looks like there are some test failures from the CI

> would this need a whatsnew entry?

Nope

**Comment by akx** [comment]
@mroeschke Rebased (#48925, #48926 conflicted...) and CI passes now.

File: pandas/tests/io/test_fsspec.py [hunk 1]
```diff
@@ -95,21 +95,19 @@ def test_to_csv_fsspec_object(cleared_fs, binary_mode, df1):
...
     path = "memory://test/test.csv"
     mode = "wb" if binary_mode else "w"
-    fsspec_object = fsspec.open(path, mode=mode).open()
-
-    df1.to_csv(fsspec_object, index=True)
-    assert not fsspec_object.closed
+    with fsspec.open(path, mode=mode).open() as fsspec_object:
+        df1.to_csv(fsspec_object, index=True)
+        assert not fsspec_object.closed
     fsspec_object.close()
```
Comment (mroeschke) [inline pandas/tests/io/test_fsspec.py:101]: These `fsspec_object.close()` shouldn't be needed with the context manager now? (Same as below)

Comment (akx) [reply pandas/tests/io/test_fsspec.py]: Good catch, fixed :)

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Awesome! Thanks @akx. Object closing PRs are very much appreciated.

**Pull request merged**